transcendence-app/src/user/infrastructure/db/postgres/user-to-role.postgres.repository.ts

```diff
@@ -0,0 +1,50 @@
+import { UserToRole, userToRoleKeys } from '../user-to-role.entity';
+import { UserWithRoles, UserWithRolesData } from '../user-with-role.entity';
+import { Injectable } from '@nestjs/common';
+import { BasePostgresRepository } from '../../../../shared/db/postgres/postgres.repository';
+import { IUserToRoleRepository } from '../user-to-role.repository';
+import { PostgresPool } from '../../../../shared/db/postgres/postgresConnection.provider';
+import { table } from '../../../../shared/db/models';
+import { makeQuery } from '../../../../shared/db/postgres/utils';
+import { userKeys } from '../user.entity';
+
+@Injectable()
+export class UserToRolePostgresRepository
+  extends BasePostgresRepository<UserToRole>
+  implements IUserToRoleRepository
+{
+  constructor(protected pool: PostgresPool) {
+    super(pool, table.USER_TO_ROLE, UserToRole);
+  }
+
+  async addUserToRole(userToRole: UserToRole): Promise<UserToRole | null> {
+    return this.add(userToRole);
+  }
+
+  async getUserWithRoles(id: string): Promise<UserWithRoles | null> {
+    const userData = await makeQuery<UserWithRolesData>(this.pool, {
+      text: `SELECT u.*,
+                    array_agg(ur.${userToRoleKeys.ROLE}) FILTER (WHERE ur.${userToRoleKeys.ROLE} IS NOT NULL) as roles
```

Comment:
```suggestion
                    coalesce( array_agg(ur.${userToRoleKeys.ROLE}) FILTER (WHERE ur.${userToRoleKeys.ROLE} IS NOT NULL), '{}' ) as roles
```

praise: I like how you implement this method and query

question: What do you think about removing the `none` role and using the [coalesce](https://www.postgresql.org/docs/current/functions-conditional.html#FUNCTIONS-COALESCE-NVL-IFNULL) function to return an empty array when the user doesn't have a role?

Reply:
done